windows/src/developer/TIKE/compile/CompileKeymanWeb.pas

```diff
@@ -873,11 +890,31 @@ function TCompileKeymanWeb.JavaScript_Name(i: Integer; pwszName: PWideChar; UseN
       else Result := '_'; // Ensures we cannot overlap numbered instances
     while pwszName^ <> #0 do
     begin
-      if CharInSet(PChar(pwszName)^, SValidIdentifierCharSet)   // I3681
-        then Result := Result + PChar(pwszName)^
-        else Result := Result + '_';
+      if CharInSet(PChar(pwszName)^, SValidIdentifierCharSet) then  // I3681
+      begin
+        Result := Result + PChar(pwszName)^
+      end
+      else
+      begin
+        Result := Result + '_';
+        FChanged := True;
+      end;
       Inc(pwszName);
     end;
+    if not UseNameForRelease then
+    begin
+      // Ensure each transformed name is still unique
+      Result := Result + '_' + IntToStr(i);
+      if FChanged then
+        Result := Result + '/*'+string(p).Replace('*/', '*-/')+'*/'
+    end
+    else if FChanged then
+    begin
+      // For named option stores, we are only supporting the valid identifier
+      // character set, which is a breaking change in 14.0.
```

Comment:
We should add this to https://github.com/keymanapp/keyman/wiki/Keyman-13.0-to-14.0-Migration#developer-and-keyboards

Reply:
So this is a new restriction on stores used with `KSAVE` and `KLOAD`, then?

If a store isn't used with a rule that forces use of its name through those two, it's still allowed to exist?

Reply:
Yes, this restriction is just on stores used with `save()` and `reset()` (aka `KSAVE` and `KLOAD` in the compiled form). I have checked, and there are no keyboards in the keymanapp/keyboards repository that fall foul of this restriction.